onyxia-api/src/main/java/fr/insee/onyxia/api/services/impl/HelmAppsService.java

```diff
@@ -416,6 +462,75 @@ private Service getHelmApp(Region region, User user, HelmLs release) {
         return service;
     }
 
+    @Override
+    public void rename(
+            Region region, Project project, User user, String serviceId, String friendlyName)
+            throws IOException, InterruptedException, TimeoutException {
+        String namespaceId =
+                kubernetesService.determineNamespaceAndCreateIfNeeded(region, project, user);
+        KubernetesClient client = kubernetesClientProvider.getUserClient(region, user);
+        Secret secret =
```

Comment:
This logic should be moved in a separate method so that it can be reused and mutualized